stats/stats/src/charts/mod.rs

```diff
@@ -128,23 +152,33 @@ pub trait ChartUpdater: Chart {
         &self,
         db: &DatabaseConnection,
         blockscout: &DatabaseConnection,
-        full: bool,
+        force_full: bool,
     ) -> Result<(), UpdateError> {
         let chart_id = crate::charts::find_chart(db, self.name())
             .await
             .map_err(UpdateError::StatsDB)?
             .ok_or_else(|| UpdateError::NotFound(self.name().into()))?;
-        let last_row = if full {
+        let min_blockscout_block = get_min_block_blockscout(blockscout)
+            .await
+            .map_err(UpdateError::BlockscoutDB)?;
+        let last_row = if force_full {
             None
         } else {
             chart_data::Entity::find()
                 .column(chart_data::Column::Date)
                 .filter(chart_data::Column::ChartId.eq(chart_id))
                 .order_by_desc(chart_data::Column::Date)
-                .into_model::<OnlyDate>()
+                .into_model::<SyncInfo>()
                 .one(db)
                 .await
                 .map_err(UpdateError::StatsDB)?
+                .filter(|row| {
+                    if let Some(block) = row.min_blockscout_block {
+                        block == min_blockscout_block
+                    } else {
+                        false
+                    }
+                })
```

Comment:
can't we filter on database query level?

Reply:
We want to check if the last row equals to this. Filtering in sql might give us a row which is equal, but old
In current implementation this shouldn't work, because all rows must have the same min block, but imo filtering in sql is more bug prone